scripts/hotrod-integration-test.sh

```diff
@@ -21,14 +46,16 @@ make create-baseimg
 
 # Loop through each platform (separated by commas)
 for platform in $(echo "$platforms" | tr ',' ' '); do
+  # Extract the operating system from the platform string
+  os=${platform%%/*}  #remove everything after the last slash
   # Extract the architecture from the platform string
   arch=${platform##*/}  # Remove everything before the last slash
-  make "build-all-in-one" GOOS=linux GOARCH="${arch}"
+  make "build-all-in-one" GOOS=${os} GOARCH="${arch}"
 done
 
 # Build image locally (-l) for integration test
-bash scripts/build-upload-a-docker-image.sh -l -c example-hotrod -d examples/hotrod -p "${platforms}"
-bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one -d cmd/all-in-one -p "${platforms}" -t release
+bash scripts/build-upload-a-docker-image.sh -l -c ${BINARY} -d examples/hotrod -p "${platforms}"
+bash scripts/build-upload-a-docker-image.sh ${LOCAL_FLAG} -b -c all-in-one -d cmd/all-in-one -p "${platforms}" -t release
```

Comment:
```suggestion
bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one -d cmd/all-in-one -p "${platforms}" -t release
```
you absolutely want to build all-in-one with `-l` - it's not even primary image for this workflow, and we definitely don't to publish it from here.

The hotrod image is also being build with `-l` (correctly) here because we have another instance of this command at the end of the script which does the publishing if the tests passed. That instance (l117) should use `${LOCAL_FLAG}` because we still don't want it to publish if it's a run from a PR